ext/pdo/pdo_dbh.c

```diff
@@ -1187,20 +1187,20 @@ PHP_METHOD(PDO, query)
 	/* we haven't created a lazy object yet */
 	ZVAL_UNDEF(&stmt->lazy_object_ref);
 
-	if (dbh->methods->preparer(dbh, statement, stmt, NULL)) {
+	if (EXPECTED(dbh->methods->preparer(dbh, statement, stmt, NULL))) {
```

Comment:
I m less convinced about this kind of case, what s the benefit here ?

Reply:
I find here and the nest cases below a bit "excessive" but that s just my opinion.

Reply:
Indeed, that the overall amount may be excessive... do you think it should be to reduce it a bit?

Reply:
I would say I have a harder time to see if there a tangible benefit for the `EXPECTED` cases (especially this suite) while most of `UNEXPECTED` look ok.

Reply:
Thank you, I will look over the unnecessary parts and fix them.